src/Features/Core/Portable/Extensions/IExtensionMessageHandlerService.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.Extensions;
+
+/// <summary>
+/// This service is used to register, unregister and execute extension message handlers.
+/// </summary>
+internal interface IExtensionMessageHandlerService : IWorkspaceService
+{
+    /// <summary>
+    /// Executes a non-document-specific extension message handler with the given message and solution.
+    /// </summary>
+    /// <param name="solution">The solution the message refers to.</param>
+    /// <param name="messageName">The name of the handler to execute. This is generally the full name of the type implementing the handler.</param>
+    /// <param name="jsonMessage">The json message to be passed to the handler.</param>
+    /// <param name="cancellationToken">Cancellation token to cancel the async operation.</param>
+    /// <returns>The json message returned by the handler.</returns>
+    ValueTask<string> HandleExtensionWorkspaceMessageAsync(
+        Solution solution,
+        string messageName,
+        string jsonMessage,
+        CancellationToken cancellationToken);
+
+    /// <summary>
+    /// Executes a document-specific extension message handler with the given message and solution.
+    /// </summary>
+    /// <param name="documentId">The document the message refers to.</param>
+    /// <param name="messageName">The name of the handler to execute. This is generally the full name of the type implementing the handler.</param>
+    /// <param name="jsonMessage">The json message to be passed to the handler.</param>
+    /// <param name="cancellationToken">Cancellation token to cancel the async operation.</param>
+    /// <returns>The json message returned by the handler.</returns>
+    ValueTask<string> HandleExtensionDocumentMessageAsync(
+        Document documentId,
+        string messageName,
+        string jsonMessage,
+        CancellationToken cancellationToken);
+
+    /// <summary>
+    /// Registers extension message handlers from the specified assembly.
+    /// </summary>
+    /// <param name="solution">The solution object.</param>
```

Comment:
remove docs that don't really tell us anything :)

Reply:
Before I make the change, do you mean that you are ok with partial xml docs? Like `summary` but no `param`s? Or some `param`s but not others?
Just making sure :-). If you are, I kind of love it <3

Reply:
Yes.  100% :)

Reply:
For example, i never want a doc comment that says "the cancellation token" :D

Reply:
Yup.  partial docs are a-ok.  we doc the interesting stuff :)